atlassian_jwt_auth/verifier.py

```diff
@@ -16,36 +16,58 @@ def __init__(self, public_key_retriever, **kwargs):
             'subject_should_match_issuer', True)
 
     def verify_jwt(self, a_jwt, audience, leeway=0, **requests_kwargs):
-        """ returns the claims of the given jwt iff verification
-            is successful.
+        """Verify if the token is correct
+
+        Returns:
+             dict: the claims of the given jwt if verification is successful.
+
+        Raises:
+            ValueError: if verification failed.
         """
+        key_identifier = key._get_key_id_from_jwt_header(a_jwt)
+        public_key = self._retrieve_pub_key(key_identifier, requests_kwargs)
+
+        return self._decode_jwt(
+            a_jwt, key_identifier.key_id, public_key,
```

Comment:
Sorry, I don't understand what should be done. Pass `key_identifie` instead of `key_identifier.key_id` to `self._decode_jwt`, or something more complex?